# HG changeset patch
# User kevinw
# Date 1375442806 -3600
#      Fri Aug 02 12:26:46 2013 +0100
# Node ID 2660219948088d89dd3fc285e093dab2520349e5
# Parent  4b3487aa553cca3fb7ecb74d284b9524d0bf5bf8
8020943: Memory leak when GCNotifier uses create_from_platform_dependent_str()
Reviewed-by: mgerdin, fparain, dcubed

diff -r 4b3487aa553c -r 266021994808 src/share/vm/services/gcNotifier.cpp
--- openjdk/hotspot/src/share/vm/services/gcNotifier.cpp	Thu Oct 17 19:28:00 2013 +0100
+++ openjdk/hotspot/src/share/vm/services/gcNotifier.cpp	Fri Aug 02 12:26:46 2013 +0100
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2011, 2012, Oracle and/or its affiliates. All rights reserved.
+ * Copyright (c) 2011, 2013, Oracle and/or its affiliates. All rights reserved.
  * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
  *
  * This code is free software; you can redistribute it and/or modify it
@@ -211,9 +211,9 @@
     NotificationMark nm(request);
     Handle objGcInfo = createGcInfo(request->gcManager, request->gcStatInfo, THREAD);
 
-    Handle objName = java_lang_String::create_from_platform_dependent_str(request->gcManager->name(), CHECK);
-    Handle objAction = java_lang_String::create_from_platform_dependent_str(request->gcAction, CHECK);
-    Handle objCause = java_lang_String::create_from_platform_dependent_str(request->gcCause, CHECK);
+    Handle objName = java_lang_String::create_from_str(request->gcManager->name(), CHECK);
+    Handle objAction = java_lang_String::create_from_str(request->gcAction, CHECK);
+    Handle objCause = java_lang_String::create_from_str(request->gcCause, CHECK);
 
     klassOop k = Management::sun_management_GarbageCollectorImpl_klass(CHECK);
     instanceKlassHandle gc_mbean_klass(THREAD, k);
